core-services/ipi-deprovision/aws.sh

```diff
@@ -3,90 +3,145 @@ set -o errexit
+      done
+
+      # remove from group
+      for group in $(aws --profile "$PROFILE" iam get-groups-for-user \
+        --user-name "$user" --query 'Groups[].GroupName' --output text 2>/dev/null); do
+          retry aws --profile "$PROFILE" iam remove-user-from-group --user-name "$user" --group-name "$group" || true
+      done
+
+      # delete user
+      if retry aws --profile "$PROFILE" iam delete-user --user-name "$user"; then
+        log "✓ Deleted IAM user: $user"
+      else
+        log "✗ Failed to delete IAM user: $user"
+      fi
+  done
+}
+
+# Route53 cleanup
+cleanup_route53() {
+  aws --profile "$PROFILE" route53 list-hosted-zones --query "HostedZones[].Id" --output text | tr '\t' '\n' | while read -r zone_id; do
+    [[ -z "$zone_id" ]] && continue
+    zone_name=$(aws --profile "$PROFILE" route53 get-hosted-zone --id "$zone_id" --query 'HostedZone.Name' --output text)
+    if [[ "$zone_name" == ci-op-* ]]; then
+      log "Cleaning Route53 zone: $zone_name ($zone_id)"
+      # delete all records except SOA/NS
+      # This is to avoid the "Resource not found" error when deleting the zone
+      records=$(aws --profile "$PROFILE" route53 list-resource-record-sets --hosted-zone-id "$zone_id" \
+        --query "ResourceRecordSets[?Type != 'SOA' && Type != 'NS']" --output json)
+      if [[ "$records" != "[]" ]]; then
+        change_batch=$(jq -c '{Changes: map({Action:"DELETE", ResourceRecordSet:.})}' <<<"$records")
+        retry aws --profile "$PROFILE" route53 change-resource-record-sets \
+          --hosted-zone-id "$zone_id" --change-batch "$change_batch" || true
+      fi
+      retry aws --profile "$PROFILE" route53 delete-hosted-zone --id "$zone_id" || true
+    fi
+  done
 }
 
-function queue() {
-	local LIVE="$(jobs | wc -l)"
-	while [[ "${LIVE}" -ge 2 ]]; do
-		sleep 1
-		LIVE="$(jobs | wc -l)"
-	done
-	echo "${@}"
-	"${@}" &
+# Hypershift cleanup
+cleanup_hypershift() {
+  log "Running Hypershift pruner..."
+  retry pruner hypershift --cutoff="$CUTOFF" --profile "$PROFILE" || true
 }
 
-function deprovision() {
-	WORKDIR="${1}"
-	REGION="$(cat ${WORKDIR}/metadata.json|jq .aws.region -r)"
-	INFRA_ID="$(cat ${WORKDIR}/metadata.json|jq '.aws.identifier[0]|keys[0]' -r|cut -d '/' -f3|tr -d '\n')"
-	if [[ -n ${HYPERSHIFT_PRUNER:-} ]]; then
-		HYPERSHIFT_BASE_DOMAIN="${HYPERSHIFT_BASE_DOMAIN:-origin-ci-int-aws.dev.rhcloud.com}"
-		timeout --signal=SIGQUIT 30m hypershift destroy infra aws --aws-creds "${AWS_SHARED_CREDENTIALS_FILE}" --infra-id "${INFRA_ID}" --base-domain "${HYPERSHIFT_BASE_DOMAIN}" --region "${REGION}" || touch "${WORKDIR}/failure"
-		timeout --signal=SIGQUIT 30m hypershift destroy iam aws --aws-creds "${AWS_SHARED_CREDENTIALS_FILE}" --infra-id "${INFRA_ID}" --region "${REGION}" || touch "${WORKDIR}/failure"
-	else
-		timeout --signal=SIGQUIT 60m openshift-install --dir "${WORKDIR}" --log-level error destroy cluster && touch "${WORKDIR}/success" || touch "${WORKDIR}/failure"
-	fi
+# VPC cleanup
```

Comment:
the vpc deletion I saw was blocked on firewalls, so you should probably check those too.